Adding new API version (2017-05-15-preview). No other changes made

finiteattractor opened the pull request:
Adding new API version (2017-05-15-preview). No other changes made, just copied all files and updated the version string.

finiteattractor committed:
Adding new API version (2017-05-15-preview). No other changes made, just copied all files and updated the version string.

viveklingaiah reviewed (changes requested):
I think you will need to another block for api 2017-05-15-preview like this file has for 2015
https://github.com/AzureAutomationTeam/azure-rest-api-specs/blob/current/specification/automation/resource-manager/readme.md

so, that you can pass an argument for 2017 and generates SDK for you. Could you also try generating SDK using autorest. Weijie did it last  week, please check with him for commands for SDK generation and linter (static validation)

finiteattractor committed:
Updating readme.md file to also include the artifacts required by new version

finiteattractor committed:
Adding -preview i missed in my last commit

finiteattractor committed:
added preview to tag line

safeermohammed commented:
@finiteattractor we can close this PR as this is not needed

finiteattractor committed:
- Removed copied files as we know how to proceed without copying.
- Small refactoring to schedule model definition so i can share it.
- Added new software update configuration specs

finiteattractor committed:
refactoring and adding examples

finiteattractor committed:
- Consolidating parameters
- Updating examples path

finiteattractor committed:
- Fixed missing closing curly braces
- Renamed some definitions to match resource
- Fixing invalid references
- Fixed example file name
- Added two resources to readme.md file

finiteattractor committed:
removed type property as we are not returning it

finiteattractor committed:
- renamed models to make more sense in generated code
- consolidated common properties (lastmodified/created-time/by)

finiteattractor committed:
Fix for wrong resource model name

safeermohammed commented on specification/automation/resource-manager/Microsoft.Automation/2015-10-31/definitions.json:
```diff
@@ -2954,6 +2954,21 @@
     },
+    "scheduleFrequency": {
+      "type": "string",
+      "description": "Gets or sets the frequency of the schedule.",
```
In last review Azure team were against using gets and sets.  Can you change the description

finiteattractor replied:
updated all! Thanks!

finiteattractor committed:
Fixed lots of example validation errors, swagger specs issues, and improved text.

viveklingaiah commented:
Hi Mo, 
   Make sure you do not have descriptions starting with "gets or sets" for definitions, they need to be more descriptive and useful. Swagger team will block, otherwise. I believe you said you've already corrected ? Also, could you replace real subscription ids in examples with some dummy id. or "subid" or all zeros ? Swagger team will raise concerns for that as well. refer to an example for accounts or runbook

viveklingaiah commented:
I can approve right now or you can change and then I approve.
let me know

finiteattractor committed:
Fixed subids in examples

finiteattractor committed:
setting x-ms-azureresource to false for non put resources
